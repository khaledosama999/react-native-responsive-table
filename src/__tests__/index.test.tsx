import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import Table from '../index'

describe('Table', () => {
  it('Renders a normal table correctly', () => {
    const normalTable = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 3</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 4</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 5</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 6</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTable).toMatchSnapshot()
  })

  it('Renders a normal table with one row correctly', () => {
    const normalTableWithOneRow = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRow).toMatchSnapshot()
  })

  it('Renders a normal table with one row and one cell correctly', () => {
    const normalTableWithOneRowAndCell = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRowAndCell).toMatchSnapshot()
  })

  it('Renders a scrollable table correctly', () => {
    const normalTable = renderer
      .create(
        <Table isScrollable={true}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 3</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 4</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 5</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 6</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTable).toMatchSnapshot()
  })

  it('Renders a normal table with overridden styles correctly', () => {
    const scrollableTable = renderer
      .create(
        <Table isScrollable={true} style={{ borderWidth: 4 }}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 3</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 4</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell>
              <Text>text 5</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 6</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(scrollableTable).toMatchSnapshot()
  })

  it('Renders a normal table with one row and overridden styles correctly', () => {
    const normalTableWithOneRowAndOverRiddenStyles = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row
            lastStyle={{ borderBottomWidth: 0 }}
            style={{ borderWidth: 5 }}
          >
            <Table.Cell>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRowAndOverRiddenStyles).toMatchSnapshot()
  })

  it('Renders a normal table with one row and cell with overridden styles correctly', () => {
    const normalTableWithOneRowAndOneCell = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell style={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRowAndOneCell).toMatchSnapshot()
  })

  it('Renders a normal table with row overridden last styles correctly', () => {
    const normalTableWithOneRowWithLastStyle = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell style={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell style={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRowWithLastStyle).toMatchSnapshot()
  })

  it('Renders a normal table with cell overridden last styles correctly', () => {
    const normalTableWithCellLastStyle = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell lastStyle={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row lastStyle={{ borderBottomWidth: 0 }}>
            <Table.Cell lastStyle={{ borderRadius: 5 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithCellLastStyle).toMatchSnapshot()
  })

  it('Renders a normal table with row overridden first styles correctly', () => {
    const normalTableWithOneRowWithLastStyle = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row firstStyle={{ borderWidth: 0 }}>
            <Table.Cell style={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell style={{ borderRadius: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithOneRowWithLastStyle).toMatchSnapshot()
  })

  it('Renders a normal table with cell overridden first styles correctly', () => {
    const normalTableWithCellLastStyle = renderer
      .create(
        <Table isScrollable={false}>
          <Table.Row>
            <Table.Cell firstStyle={{ borderWidth: 4 }}>
              <Text>text 1</Text>
            </Table.Cell>
            <Table.Cell>
              <Text>text 2</Text>
            </Table.Cell>
          </Table.Row>
        </Table>
      )
      .toJSON()

    expect(normalTableWithCellLastStyle).toMatchSnapshot()
  })
})
