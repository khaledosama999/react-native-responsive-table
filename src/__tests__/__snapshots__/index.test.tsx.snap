// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Renders a normal table correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 2
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 3
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 4
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 6
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with cell overridden first styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 4,
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with cell overridden last styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with one row and cell with overridden styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with one row and one cell correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with one row and overridden styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "borderWidth": 5,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with one row correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": "50%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with overridden styles correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 4,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 1
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 3
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 4
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 0,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 6
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Table Renders a normal table with row overridden first styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "borderWidth": 0,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a normal table with row overridden last styles correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "grey",
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "display": "flex",
          "flexBasis": "100%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text>
        text 1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Table Renders a scrollable table correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "grey",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": undefined,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 1
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 3
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 4
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 0,
          "borderColor": "grey",
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 5
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexBasis": "50%",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingHorizontal": 4,
            "paddingVertical": 4,
          }
        }
      >
        <Text>
          text 6
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
